dlpy/image_captioning.py

```diff
@@ -0,0 +1,551 @@
+        else:
+            # resize images for object detection scoring
+            image_table.resize(objects_width)
+
+            detected_objects = create_embeddings_from_object_detection(conn,image_table,obj_detect_model,word_embeddings_file,word_delimiter=embeddings_delimiter,
+                                                      n_threads=n_threads,gpu=gpu)
+            # conn.dljoin(table=dict(name='captions_features'),annotatedTable=detected_objects,
+            #             id='_filename_0',casOut=dict(name='obj_capt_feats',replace=True))
+            df1 = detected_objects.to_frame()
+            df2 = result.to_frame()
+            obj_capt_feat = pd.merge(df1,df2,left_on='_filename_0',right_on='_filename_0',how='left')
+            result = conn.upload_frame(obj_capt_feat,casout=dict(name='full_table',replace=True))
+
+    final_table = reshape_caption_columns(conn,result,caption_col_name=caption_col_name,num_captions=num_captions)
+    drop_columns = set(final_table.columns) - set(captions_table.columns) - set(image_features.columns)
+    if obj_detect_model:
+        drop_columns = set(drop_columns) - set(detected_objects.columns)
+    drop_columns.remove('caption')
+    final_table.drop(drop_columns,axis=1,inplace=True)
+
+    return final_table
+
+def ImageCaptioning(conn,model_name='image_captioning',num_blocks=3,neurons=50,
+                    rnn_type='LSTM',max_output_len=15):
+    '''
+    Builds an RNN to be used for image captioning
+
+    Parameters
+    ----------
+    conn : CAS
+        Specifies the CAS connection object.
+    model_name : string, optional
+        Specifies output name of the model
+        Default: 'image_captioning'
+    num_blocks : int, optional
+        Specifies number of samelength recurrent layers
+        Default : 3
+    neurons : int, optional
+        Specifies number of neurons in each layer
+        Default : 50
+    rnn_type : string, optional
+        Specifies the type of the rnn layer. Possible Values: RNN, LSTM, GRU
+        Default: LSTM
+    max_output_len : int, optional
+        Specifies max number of tokens to generate in the final layer (i.e. max caption length)
+        Default : 15
+    Returns
+    -------
+    CASTable
+
+    '''
+    model = Sequential(conn, model_table=model_name)
+
+    model.add(InputLayer(name='input'))
+    print('InputLayer added named "input"')
+    for i in range(num_blocks):
+        model.add(Recurrent(n=neurons, init='msra', rnn_type=rnn_type, output_type='samelength'))
+
+    model.add(Recurrent(n=neurons, init='msra', rnn_type=rnn_type, output_type='encoding'))
+
+    model.add(Recurrent(n=neurons, init='msra', rnn_type=rnn_type, output_type='arbitrarylength',
+                        max_output_length=max_output_len))
+
+    model.add(OutputLayer(name='output'))
+    print('OutputLayer added named "output"')
+    return model
+
+def show_predictions(conn,result_tbl,npreds=2,ncol=2,img_path=None,figsize=None):

```

Comment:
lets change this to display_image_captions or display_predicted_image_captions

Reply:
can you use this function to visualize the training set? if so maybe we can call it display_image_captions